go.mod

```diff
@@ -1,6 +1,8 @@
 module sigs.k8s.io/kubebuilder-declarative-pattern
 
-go 1.20
+go 1.21
+
+toolchain go1.22.0
```

Comment:
Maybe, we should update the files for GitHub Actions?
- https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/.github/workflows/main.yml
This file use the go v1.20 setup at 2 place.

Reply:
You're right - I just pushed that and tests are looking much happier!

Reply:
And I just saw that @tomasaschan proposed a better version of the change in https://github.com/justinsb/kubebuilder-declarative-pattern/pull/1 , so using that one